File: src/core/path-entry.ts
/** PATH 路径条目 — 包含路径值和启用状态 */
export interface PathEntry {
  path: string;
  enabled: boolean;
}

/** 系统 PATH 与用户 PATH 的完整有序快照。 */
export interface PathSnapshot {
  system: PathEntry[];
  user: PathEntry[];
}

File: core/src/path_entry.rs
use serde::{Deserialize, Serialize};

/// PATH 路径条目 — Rust 与 TypeScript 共享的领域对象。
#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
pub struct PathEntry {
    pub path: String,
    pub enabled: bool,
}

/// 系统 PATH 与用户 PATH 的一致快照。
#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize, Default)]
pub struct PathSnapshot {
    #[serde(default)]
    pub system: Vec<PathEntry>,
    #[serde(default)]
    pub user: Vec<PathEntry>,
}
